sources/standard/filesystem/__init__.py

```diff
@@ -0,0 +1,178 @@
+"""This source collects fsspec files and downloads or proccess them."""
+import mimetypes
+import os
+from typing import Optional
+
+import dlt
+from dlt.common import pendulum
+from dlt.common.storages import filesystem
+from dlt.common.storages.configuration import FilesystemConfiguration
+from dlt.common.time import ensure_pendulum_datetime
+from dlt.extract.source import DltResource, TDataItem, TDataItems
+
+from ..file_source import FileModel
+from .settings import DEFAULT_CHUNK_SIZE, DEFAULT_START_DATE, STORAGE_PATH
+
+from fsspec import AbstractFileSystem
+import pandas as pd
+
+
+@dlt.source
+def filesystem_source(
+    credentials: FilesystemConfiguration = dlt.secrets.value,
```

Comment:
I'd like to have something more user friendly: just pass bucket_url and credentials like you see them `FilesystemConfiguration` as arguments and use `FilesystemConfiguration` as spec in `dlt.resource`. (you did that in Mondo).
in next step we'll subclass  `FilesystemConfiguration` and add `storage_path` `start_date` etc. to it